app/labor/rate_limit_checker.rb

```diff
@@ -22,11 +38,13 @@ def limit_by_action(action)
   end
 
   def track_image_uploads
-    Rails.cache.increment("#{@user.id}_image_upload", 1, expires_in: 30.seconds)
+    expires_in = RETRY_AFTER[:image_upload].seconds
+    Rails.cache.increment("#{@user.id}_image_upload", 1, expires_in: expires_in)
```

Comment:
Hi @rhymes, I would like to report a problem here:
Rails version 5.2.4.2 (latest version of 5.2) does not support `expires_in` for `Rails.cache.increment` for redis_cache_store. As a result, value will be expired after 1 hour (by default), not 30 seconds as expected.
https://github.com/rails/rails/blob/v5.2.4.2/activesupport/lib/active_support/cache/redis_cache_store.rb#L262
This was fixed in master branch:
https://github.com/rails/rails/blob/master/activesupport/lib/active_support/cache/redis_cache_store.rb#L277
Please help to correct me if I'm wrong...

Reply:
@tamcv omg amazing find! Though our default expiration is 24 hours, not 1 hour:

https://github.com/thepracticaldev/dev.to/blob/1e5d831a3ed6339e9886c8300d418bd7bce13ac0/config/environments/production.rb#L115-L116


I tested with:

```ruby
[3] pry(main)> Rails.cache.increment("rhymes_counter_x", 1, expires_in: 45.seconds)
Cache increment: rhymes_counter_x ({:amount=>1})
=> 1
```

```shell
➜  ~ redis-cli get rhymes_counter_x && redis-cli ttl rhymes_counter_x
"1"
(integer) -1
```

and indeed there's no expiration date :( 

I'm going to merge this as is a it also refactors the code, would you like to open a separate bug issue about this and propose a solution? I personally would like to keep the atomic `INCRBY` separation but we can't upgrade to Rails 6 just yet, maybe there's a way to have both the cake and eat it too :D

Reply:
ah @tamcv, a possible avenue (thanks @maestromac for the suggestion) would be move these rate limits up to `rack-attack`, see https://github.com/thepracticaldev/dev.to/blob/master/config/initializers/rack/attack.rb

what do you think?

Reply:
Yes @rhymes , Rack::Attack is the correct way to throttle the requests.

Reply:
@rhymes I create new issue here: https://github.com/thepracticaldev/dev.to/issues/7503
In my opinion, we might need to clone the patch instead of moving the limit to `rack-attack` because `rack-attack` throttle the requests based on `ip_address`, there is no `user_id` at that step...